bmaptools/BmapCopy.py

```diff
@@ -211,6 +213,20 @@ def __init__(self, image, dest, bmap=None, image_size=None):
 
         self._batch_blocks = self._batch_bytes // self.block_size
 
+    def set_psplash_pipe(self, pipe):
+        """
+        Set the psplash named pipe file path to be used when updating the
+        progress.
+
+        The 'pipe' argument is the named pipe used by the psplash process to get
+        progress bar commands.
+        """
+        if os.path.exists(pipe) and stat.S_ISFIFO(os.stat(pipe).st_mode):
+            self.psplash_pipe = pipe
+        else:
+            _log.warning("%s is not a pipe so psplash progress will not be "
+                         "updated" % pipe)
```

Comment:
I think this should be an error, not warning.

If I use this tool, and I need the progress, and I tell the tool to pipe it, I do mean it, I do need it. I do not want the tool just silently ignore my need with a warning I may never see, and continue. I want the tool to promptly fail.

I think this is a good default strategy for all options. There may be exceptions, is this the exceptional case?

Reply:
In general, I agree. The issue here is that this is meant to be part of a boot sequence where not having psplash (or not having a display) should not affect the functional writing process. Its semantics would be along the "update status if available" lines. Think about a boot process that writes a disk and updates a graphical interface when available. You wouldn't want to break the writing process if there is no display attached.

Reply:
OK, in this case please, make it clear that this is best effort in the help text, thanks.

Reply:
Will also be helpful to mentions this in the docstring, along with motivation why, just in case someone some day wonders. May be in commit message :) Now I feel I sound paranoid :)

Reply:
No. It's absolutely reasonable and I will make it clear.

Reply:
The should have clarified this. Closing this thread but feel free to comment if needs anything more.